Sources/core/objects/Tables.cpp

```diff
@@ -0,0 +1,319 @@
+#include "Tables.h"
+#include "Item.h"
+#include "Human.h"
+
+Table::Table(size_t id) : IMovable(id)
+{
+    anchored = true;
+    v_level = 5;
+    SetPassable(D_ALL, Passable::SMALL_CREATURE);
+    SetSprite(""); 
+    material_ = "";
+    SetState(material_ + "_table_d0");
+}
+
+void Table::AfterWorldCreation()
+{
+    NotifyNeighborhood(true);
+}
+
+void Table::Delete()
+{
+    NotifyNeighborhood(false);
+    IOnMapObject::Delete();
+}
+void Table::NotifyNeighborhood(bool is_in_existence)
+{
+    size_t id;
+    if(is_in_existence)
+    {
+        id = 0;
+    }
+    else
+    {
+        id = GetId();
+    }
+    if (auto table_up = GetNeighbour(D_UP)->GetItem<Table>())
+    {
+        table_up->UpdateSprite(id);
+    }
+    if (auto table_down = GetNeighbour(D_DOWN)->GetItem<Table>())
+    {
+        table_down->UpdateSprite(id);
+    }
+    if (auto table_left = GetNeighbour(D_LEFT)->GetItem<Table>())
+    {
+        table_left->UpdateSprite(id);
+    }
+    if (auto table_right = GetNeighbour(D_RIGHT)->GetItem<Table>())
+    {
+        table_right->UpdateSprite(id);
+    }
+    if (auto table_upleft = GetNeighbour(D_LEFT)->GetNeighbour(D_UP)->GetItem<Table>())
+    {
+        table_upleft->UpdateSprite(id);
+    }
+    if (auto table_downleft = GetNeighbour(D_LEFT)->GetNeighbour(D_DOWN)->GetItem<Table>())
+    {
+        table_downleft->UpdateSprite(id);
+    }
+    if (auto table_upright = GetNeighbour(D_UP)->GetNeighbour(D_RIGHT)->GetItem<Table>())
+    {
+        table_upright->UpdateSprite(id);
+    }
+    if (auto table_downright = GetNeighbour(D_DOWN)->GetNeighbour(D_RIGHT)->GetItem<Table>())
+    {
+        table_downright->UpdateSprite(id);
+    }
+    UpdateSprite(id);
+}
+void Table::UpdateSprite(size_t ignored_table)
+{
+    bool up  = false;
+    bool down = false;
+    bool left = false;
+    bool right = false;
+    bool upright = false;
+    bool downright = false;
+    bool upleft = false;
+    bool downleft = false;
+    if (bool(GetNeighbour(D_UP)->GetItem<Table>()) != ignored_table && GetNeighbour(D_UP)->GetItem<Table>())
+    {
```

Comment:
1) You are casting `IdPtr` to `bool` type and after that compare it with `size_t` type. It will not work properly, because it is needed to compare id of the objects.
```
GetNeighbour(D_UP)->GetItem<Table>().Id() != ignored_table
```
2) It is better to create some helper function here. Something like that:
```
int CheckTable(IdPtr<IOnMapBase> container, size_t ignored_table)
{
    IdPtr<Table> table = container->GetItem<Table>();
    if (!table.IsValid())
    {
        return 0;
    }
    if (table.Id() == ignored_table)
    {
        return 0;
    }
    return 1;
}
```
Usage:
```
int up = CheckTable(GetNeighbour(D_UP), ignored_table);
```
3) Instead of `bool` for `up`, `down`, etc use `int`. You use addition on them, and it is not really good to perform any arithmetic operations with `bool` variables.